src/power_grid_model/_core/power_grid_meta.py

```diff
@@ -197,11 +200,29 @@ def initialize_array(
     component_type = _str_to_component_type(component_type)
     if not isinstance(shape, tuple):
         shape = (shape,)
-    if empty:
-        return np.empty(shape=shape, dtype=power_grid_meta_data[data_type][component_type].dtype, order="C")
-    return np.full(
-        shape=shape,
-        fill_value=power_grid_meta_data[data_type][component_type].nan_scalar,
-        dtype=power_grid_meta_data[data_type][component_type].dtype,
-        order="C",
-    )
+    component_meta = power_grid_meta_data[data_type][component_type]
+    if attributes is None:
+        if empty:
+            return np.empty(shape=shape, dtype=component_meta.dtype, order="C")
+        return np.full(
+            shape=shape,
+            fill_value=component_meta.nan_scalar,
+            dtype=component_meta.dtype,
+            order="C",
+        )
+    data = {}
+    for attribute in attributes:
+        if component_meta.dtype.names is None:
+            raise PowerGridUnreachableHitError
+        if attribute not in component_meta.dtype.names:
+            raise ValueError(f"Attribute {attribute} is not available for {component_type}")
+        if empty:
+            data[attribute] = np.empty(shape=shape, dtype=component_meta.dtype[attribute], order="C")
+        else:
+            data[attribute] = np.full(
+                shape=shape,
+                fill_value=component_meta.nan_scalar[attribute],
+                dtype=component_meta.dtype[attribute],
+                order="C",
+            )
```

Comment:
do you think it makes to put this in a (private) function since it's the same logic as before but with a different dtype?

Reply:
Thought about it, as it is all same logic. But they are essentially all single lines.
I think it would not benefit readability much. Hence kept it as such